railties/test/commands/server_test.rb

```diff
@@ -108,4 +108,14 @@ def test_log_stdout
       end
     end
   end
+
+  def test_default_options
+    server = Rails::Server.new
+    old_default_options = server.default_options
+
+    # Regression test for issue #22811
```

Comment:
@eileencodes  Thanks for your feedback. I removed the comment, rebased onto master, and pushed the update to this PR.